packages/widget/src/components/common/dropdown/dropdown.ts

```diff
@@ -34,7 +35,10 @@ export class Dropdown extends BaseComponent {
   label? = '';
 
   @property({ type: Array })
-  options: DropdownOption[] = [];
+  options: DropdownOption[] | HTMLTemplateResult = [];
+
+  @property({ type: Object })
+  actionOption?: HTMLTemplateResult | string | null = null;
```

Comment:
I'd leave it like this and we can update in future if necessary

```suggestion
  actionOption?: HTMLTemplateResult | null = null;
```